atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/metrics/SweepOutcomeMetrics.java

```diff
@@ -41,19 +42,24 @@ private SweepOutcomeMetrics() {
 
     public static SweepOutcomeMetrics registerLegacy(MetricsManager metricsManager) {
         SweepOutcomeMetrics metrics = new SweepOutcomeMetrics();
-        metrics.registerMetric(metricsManager, LEGACY_OUTCOMES, BackgroundSweeperImpl.class);
+        metrics.registerMetric(metricsManager, LEGACY_OUTCOMES, ImmutableMap.of(), BackgroundSweeperImpl.class);
         return metrics;
     }
 
-    public static SweepOutcomeMetrics registerTargeted(MetricsManager metricsManager) {
+    public static SweepOutcomeMetrics registerTargeted(MetricsManager metricsManager, Map<String, String> strategyTag) {
         SweepOutcomeMetrics metrics = new SweepOutcomeMetrics();
-        metrics.registerMetric(metricsManager, TARGETED_OUTCOMES, TargetedSweepMetrics.class);
+        metrics.registerMetric(metricsManager, TARGETED_OUTCOMES, strategyTag, TargetedSweepMetrics.class);
         return metrics;
     }
 
-    private void registerMetric(MetricsManager manager, List<SweepOutcome> outcomes, Class<?> forClass) {
+    private void registerMetric(MetricsManager manager, List<SweepOutcome> outcomes,
+            Map<String, String> additionalTags, Class<?> forClass) {
         outcomes.forEach(outcome -> manager.registerOrGet(forClass, AtlasDbMetricNames.SWEEP_OUTCOME,
-                () -> getOutcomeCount(outcome), ImmutableMap.of(AtlasDbMetricNames.TAG_OUTCOME, outcome.name())));
+                () -> getOutcomeCount(outcome),
+                ImmutableMap.<String, String>builder()
+                        .putAll(additionalTags)
```

Comment:
think this might scream at you if there are duplicates, not that I imagine there would be any but you know 🤷‍♂

Reply:
True, but if we get to that I want it to scream, as something went badly wrong :P